### PR TITLE
migrate assignment permissions MODPERMS-160

### PR DESCRIPTION
**Pull request by adamdickmeiss**
https://issues.folio.org/browse/MODPERMS-160

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java [hunk 1]
```diff
@@ -145,16 +143,14 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
           return true;
         })
         .forEach(newPerms::add);
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:145]:
```suggestion
        .collect(Collectors.toList());
```
avoids newPerms

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java [hunk 2]
```diff
@@ -169,32 +165,21 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
...
-            return dbPerms.containsKey(name)
-                && (dbPerms.get(name).getDeprecated()
-                || !PermissionUtils.equals(okapiPerm, moduleId.getProduct(), dbPerms.get(name)));
-          })
+          String name = okapiPerm.getPermissionName();
+          return dbPerms.containsKey(name)
+              && (dbPerms.get(name).getDeprecated()
+              || !PermissionUtils.equals(okapiPerm, moduleId.getProduct(), dbPerms.get(name)));
+        })
         .forEach(modifiedPerms::add);
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:182]:
```
        .collect(Collectors.toList());
```
avoids modifiedPerms and is more stream-like.

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java [hunk 3]
```diff
@@ -204,15 +189,15 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
...
 
     perms.stream().forEach(perm -> permNames.add(perm.getPermissionName()));
-
     dbPerms.values().stream()
-      .filter(dbPerm -> !permNames.contains(dbPerm.getPermissionName()))
-      .forEach(removedPerms::add);
+        .filter(dbPerm -> !permNames.contains(dbPerm.getPermissionName()))
+        .forEach(removedPerms::add);
 
     return removedPerms;
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:196]: Set<String> permNames = perms.stream()
        .map(OkapiPermission::getPermissionName)
        .collect(Collectors.toSet());

    return dbPerms.values().stream()
      .filter(dbPerm -> !permNames.contains(dbPerm.getPermissionName()))
      .collect(Collectors.toList());

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java [hunk 4]
```diff
@@ -226,28 +211,79 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
...
+              permissionsToBeAdded = Collections.emptyList();
+            }
+            boolean added = false;
+            for (String perm : permissionsToBeAdded) {
+              if (!originalList.contains(perm)) {
+                newList.add(perm);
+                added = true;
+              }
+            }
+            if (added) {
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:254]:
```suggestion
            if (originalList.contains(PermissionUtils.PERMS_OKAPI_ALL)) {
              newList.add(PermissionUtils.PERMS_USERS_ASSIGN_OKAPI);
              newList.add(PermissionUtils.PERMS_USERS_ASSIGN_IMMUTABLE);
              newList.add(PermissionUtils.PERMS_USERS_ASSIGN_MUTABLE);
            } else if (originalList.contains(PermissionUtils.PERMS_PERMS_ALL)) {
              newList.add(PermissionUtils.PERMS_USERS_ASSIGN_IMMUTABLE);
              newList.add(PermissionUtils.PERMS_USERS_ASSIGN_MUTABLE);
            }
            if (newList.size() > originalList.size()) {
```

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java [hunk 5]
```diff
@@ -226,28 +211,79 @@ public void postTenantpermissions(OkapiPermissionSet entity, Map<String, String>
...
+    return savePermList(moduleId, modifiedPerms, connection, vertxContext, tenantId);
+  }
+
+  private Future<Void> handleRenamedPerms(ModuleId moduleId,  Map<String,Permission> dbPerms, @NotNull List<OkapiPermission> permList,
+      Conn connection, Context vertxContext, String tenantId) {
+
+    Map<OkapiPermission, List<Permission>> renamedPerms = getRenamedPerms(dbPerms, permList);
+    if (renamedPerms.isEmpty()) {
       return Future.succeededFuture();
     }
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:285]: This if can go into renamePermList.

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 4 above)
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/TenantPermsAPI.java:254]: I prefer
`if (newList.size() > originalList.size()) {`

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/TenantPermsAPI.java:145]: nice one

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/TenantPermsAPI.java:196]: ok

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/TenantPermsAPI.java:182]: ok

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 5 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/TenantPermsAPI.java:285]: ok

File: src/main/java/org/folio/rest/impl/TenantPermsAPI.java (hunk 4 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/TenantPermsAPI.java:254]: That would work.. But I think current behavior is more clearer.

**Pull request merged**